carta/session.py

```diff
@@ -358,39 +359,58 @@ def set_starting_directory(self, path):
+        """Open or append a new image.
 
         Parameters
         ----------
         path : {0}
             The path to the image file, either relative to the session's current directory or an absolute path relative to the CARTA backend's root directory.
         hdu : {1}
             The HDU to select inside the file.
-        complex : {2}
-            Whether the image is complex. Set to ``False`` by default.
-        expression : {3}
-            Arithmetic expression to use if opening a complex-valued image. The default is :obj:`carta.constants.ArithmeticExpression.AMPLITUDE`.
+        append : {2}
+            Whether the image should be appended to existing images. By default this is ``False`` and any existing open images are closed.
         """
-        return Image.new(self, path, hdu, False, complex, expression)
+        path = self.resolve_file_path(path)
+        directory, file_name = posixpath.split(path)
+        return Image.new(self, directory, file_name, hdu, append, False)
 
-    @validate(String(), String(r"\d*"), Boolean(), Constant(ArithmeticExpression))
-    def append_image(self, path, hdu="", complex=False, expression=ArithmeticExpression.AMPLITUDE):
-        """Append a new image, keeping any existing images.
+    @validate(String(), Constant(ComplexExpression), Boolean())
+    def open_complex_image(self, path, expression=ComplexExpression.AMPLITUDE, append=False):
+        """Open or append a new complex-valued image.
 
         Parameters
         ----------
         path : {0}
-            The path to the image file, either relative to the session's current directory or an absolute path relative to the CARTA backend's root directory.
-        hdu : {1}
-            The HDU to select inside the file.
-        complex : {2}
-            Whether the image is complex. Set to ``False`` by default.
-        expression : {3}
-            Arithmetic expression to use if appending a complex-valued image. The default is :obj:`carta.constants.ArithmeticExpression.AMPLITUDE`.
-        """
-        return Image.new(self, path, hdu, True, complex, expression)
+            The path to the complex-valued image file, either relative to the session's current directory or an absolute path relative to the CARTA backend's root directory.
+        expression : {1}
+            The complex expression to use when opening the image. The default is :obj:`carta.constants.ComplexExpression.AMPLITUDE`.
+        append : {2}
+            Whether the image should be appended to existing images. By default this is ``False`` and any existing open images are closed.
+        """
+        path = self.resolve_file_path(path)
+        directory, file_name = posixpath.split(path)
+        expression = f'{expression}("{file_name}")'
+        return Image.new(self, directory, expression, "", append, True)
+
+    @validate(String(), String(), Boolean())
+    def open_LEL_image(self, expression, directory=None, append=False):
```

Comment:
Once my changes are merged in, the default for `directory` here should be changed to `""` or `"."`, and the explicit handling of `None` in the body of the function should be removed. (This is in addition to moving the `resolve_file_path` step back into `Image.new`.)